idseq_pipeline/commands/host_filtering_functions.py

```diff
@@ -352,8 +362,10 @@ def run_star(fastq_files):
     def unmapped_files_in(some_dir):
         return ["%s/Unmapped.out.mate%d" % (some_dir, i+1) for i in range(num_fastqs)]
     genome_dir = fetch_genome(STAR_GENOME)
-    # If we are here, we are also going to need a bowtie genome later;  start fetching it now
+    # If we are here, we are also going to need a bowtie genome and gsnap genome later;  start fetching it now
     threading.Thread(target=fetch_genome, args=[BOWTIE2_GENOME]).start()
+    if check_s3_file_presence(GSNAP_GENOME):
```

Comment:
Could you launch this in Bowtie2?   Where it does
```
fetch_genome(bowtie2)
```
change it to
```
fetch_genome(bowtie2)
if check_s3_file_presence(GSNAP_GENOME):
    threading.Thread(target=fetch_genome, args=[GSNAP_GENOME]).start()
```
The reason is that s3mi is pretty good at utilizing the machine resources and you don't want to have 2 semis running in parallel with star.   Doing as I suggest will ensure only 1 semi at a time.  And it's fine to have it fetch while bowtie is running.